index.html

```diff
@@ -916,6 +920,7 @@ <h3>
           attribute EventHandler onicecandidatepairremove;
           attribute EventHandler onicecandidatepairnominate;
           undefined setSelectedCandidatePair(RTCIceCandidatePair candidatePair);
+          Promise&lt;undefined&gt; removeCandidatePairs(sequence&lt;RTCIceCandidatePair&gt; candidatePairs);
```

Comment:
Why does it take an array?

What's wrong with:
```js
await Promise.all(pairs.map(pair => transceiver.transport.iceTransport.removeCandidatePair(pair)));
```

Reply:
The array input is meant as an optimization for a few reasons.

Removing candidate pairs can potentially generate a lot of activity:
- a remove event for each removed pair
- state change events if the currently selected candidate pair is removed

With array, there is also the possibility for an application to select a candidate pair and then remove all unused pairs in a single operation. In general, removing several pairs can be expected to be a common operation when many remove events are cancelled and/or after the application selects a pair for good.

Reply:
> The array input is meant as an optimization for a few reasons.

What needs to be optimized?

> Removing candidate pairs can potentially generate a lot of activity:
> 
>     * a remove event for each removed pair

We either want the events or not, and if we do, we should fire them individually I think, and not entertain an (array) API that might get us into considering coalescing events on a single task. I think that only complicates matters and makes this harder for apps to reason about.

If we don't want events, maybe we shouldn't fire them for JS-initiated actions? (I believe I've questioned this elsewhere).

>     * state change events if the currently selected candidate pair is removed

There's only one selected pair, so I don't see what an array optimizes. 

> With array, there is also the possibility for an application to select a candidate pair and then remove all unused pairs in a single operation. In general, removing several pairs can be expected to be a common operation when many remove events are cancelled and/or after the application selects a pair for good.

Wht does "for good" mean? Is it
1. let nomination complete, or
2. remove all pairs but the selected pair?

If it's 2 then what happens to connectivity checks? Do they continue, wasting resources?

I'm concerned this API is a workaround for the "never nominate" hack that might leave apps to different ways to conclude ICE, with one draining more resources than the other.

Reply:
> If we don't want events, maybe we shouldn't fire them for JS-initiated actions?

We want all events to fire for a consistent API - all candidate pair additions are announced by the `icecandidatepairadd` event and removals announced by the `icecandidatepairremove` event, and always after the ICE agent has updated its internal state. I think selectively not firing a subset of events would make the API clunkier, even if all capability was retained (of which I am not completely certain in any case).

> If it's 2 then what happens to connectivity checks? Do they continue, wasting resources?

Removing candidate pairs will stop connectivity checks and release their resources. That is the explicit goal of this API and I've made edits to clarify that in the latest commit.

> I'm concerned this API is a workaround for the "never nominate" hack that might leave apps to different ways to conclude ICE, with one draining more resources than the other.

Giving apps more control over ICE is the central mission of the new APIs. Yes, that might mean using more resources if, for instance, the app thinks it will improve the reliability of the call. An app could also drain resources by creating tons of other peer connections, so I'm not sure I follow how this API makes it possible to drain more resources than many other options that are already possible today.

Having said that, changing the API to take a single argument instead of an array doesn't make any functional difference to the capability of the API. I can change it to a single argument and we could revisit and add an array overload in the future if it seems useful.